hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/HazelcastWriters.java

```diff
@@ -55,6 +59,90 @@
+                serializableConfig(clientConfig),
+                index -> new ArrayMap(),
+                ArrayMap::add,
+                instance -> {
+                    IMap map = instance.getMap(name);
+                    Map<K, V> tmpMap = new HashMap<>();
+                    AbstractEntryProcessor entryProcessor = new AbstractEntryProcessor() {
+                        @Override
+                        public Object process(Entry entry) {
+                            entry.setValue(updateFn.apply(entry, tmpMap.get(entry.getKey())));
+                            return null;
+                        }
+                    };
+
+                    return buffer -> {
+                        try {
+                            if (buffer.isEmpty()) {
+                                return;
+                            }
+                            for (Entry<Object, Object> entry : buffer.entrySet()) {
+                                // on duplicate key, we'll flush immediately
+                                if (tmpMap.containsKey(entry.getKey())) {
+                                    map.executeOnKeys(tmpMap.keySet(), entryProcessor);
+                                    tmpMap.clear();
+                                }
+                                tmpMap.put((K) entry.getKey(), (V) entry.getValue());
+                            }
+                            map.executeOnKeys(tmpMap.keySet(), entryProcessor);
+                            tmpMap.clear();
+                        } catch (HazelcastInstanceNotActiveException e) {
+                            handleInstanceNotActive(instance, e, isLocal);
+                        }
+                        buffer.clear();
+                    };
+                },
+                noopConsumer()
+        ));
+    }
+
+    @Nonnull
+    @SuppressWarnings("unchecked")
+    public static <T, K, V> ProcessorMetaSupplier updateMapP(
+            String name,
+            ClientConfig clientConfig,
+            DistributedFunction<T, K> toKeyFn,
+            DistributedFunction<T, EntryProcessor<K, V>> toEntryProcessorFn
+    ) {
+        boolean isLocal = clientConfig == null;
+        return dontParallelize(new HazelcastWriterSupplier<>(
+                serializableConfig(clientConfig),
+                index -> new ArrayMap(),
+                ArrayMap::add,
+                instance -> {
+                    IMap map = instance.getMap(name);
+                    return buffer -> {
+                        try {
+                            if (buffer.isEmpty()) {
+                                return;
+                            }
+                            for (Entry<Object, Object> entry : buffer.entrySet()) {
+                                EntryProcessor<K, V> entryProcessor = toEntryProcessorFn.apply((T) entry);
+                                K key = toKeyFn.apply((T) entry);
+                                map.executeOnKey(key, entryProcessor);
```

Comment:
This will be very slow, it's processing entries one by one synchronously, with 1ms network latency 500 keys/s at most... If we keep this version, performance penalty should be at least document. But I think we can adapt all use cases of this version to the other version, which works in batches, so there's no need to keep this version

Reply:
I know it's performance drawbacks. Let me discuss about the requirements with Vladimir and Can whether to keep this version or not. I agree with your point that if we keep this, we need to make aware the users that it makes a remote call for each item in the stream.

Reply:
IMap also has `    ICompletableFuture submitToKey(K key, EntryProcessor entryProcessor);` cant you use this?

Reply:
It will improve the latency but doesn't change the fact that we are making a network call per item. I am not strongly opinionated about this, so we could use that. The main discussion should be whether we need to support user provided entryprocessors or force users to implement the logic in the update functions.